add apple silicon section (#827)
void-linux/void-docs: src/installation/guides/arm-devices/platforms.md

This breaks workloads which use containers. Therefore, if you want to use
containers on your Raspberry Pi, you need to enable memory cgroups by adding
`cgroup_enable=memory` to `/boot/cmdline.txt`.

## Apple Silicon

> For some time, the kernel, mesa and uboot packages will have Asahi Linux as

Reviewer: ```suggestion
For some time, the kernel, mesa, and uboot packages will have Asahi Linux as
```

upstream. Their [documentation](https://github.com/AsahiLinux/docs/wiki) is an
excellent source. Here we discuss Void Linux specific aspects only.

Use Asahi Linux install scripts with `UEFI environment only`, then proceed with
usual Void installation. The default is EFI partition at `/boot/efi`, other
choices can be made by editing m1n1 kernel hook config.

> At the very least, install `base-system`, `asahi-base` and an initramfs

Reviewer: ```suggestion
At the very least, install `base-system`, `asahi-base`, and an initramfs
```

> generator. By default, also `asahi-uboot`, `grub-arm64-efi` and `asahi-scripts`
> are required, unless one changes initramfs or payload for m1n1.

Reviewer: ```suggestion
generator. By default, `asahi-uboot`, `grub-arm64-efi`, and `asahi-scripts`
are also required, unless one changes initramfs or payload for m1n1.
```

> Void ships a kernel hook for m1n1, modeled upon Asahi upstream. This allows to

Reviewer: ```suggestion
Void ships a kernel hook for m1n1, modeled on Asahi upstream. This allows you to
```

either just use m1n1, or use it in combination with uboot and grub (default) by

> setting the `PAYLOAD` variable (requires `asahi-uboot`, `grub-arm64-efi`). We

Reviewer: ```suggestion
setting the `PAYLOAD` variable (requires `asahi-uboot` and `grub-arm64-efi`). We
```

ship tinyramfs config and hooks (to use, add `asahi` to `hooks`). The default is
dracut, its hooks are provided by `asahi-scripts` together with mkinitcpio ones.

If using audio, you must install `asahi-audio`, ensure the speakersafetyd

> service is enabled, and utilize pipewire and wireplumber.

Reviewer: ```suggestion
service is [enabled](/config/services/index.md#enabling-services), and set up [pipewire and wireplumber](/config/media/pipewire.md).
```

Use Asahi Linux install scripts with `UEFI environment only`. U-boot should pick
up the external USB by default, in case it does not the commands

```
setenv boot_targets "usb"
setenv bootmeths "efi"
boot
```

will instruct it to do so. Then proceed with the usual Void installation. At the
very least, install `base-system`, `asahi-base`, and an initramfs generator. By
default, `grub-arm64-efi` and `asahi-scripts` are also required. The latter
provides hooks for dracut and mkinitcpio. Void also ships tinyramfs config and
hooks (to use, add `asahi` to `hooks`). The grub option `--removable` is known
to work.

Optionally, you can install `m1n1`. Void ships a kernel hook for m1n1, modeled
upon Asahi upstream. This allows you to either just use m1n1, or use it in
combination with uboot and grub by setting the `PAYLOAD` variable (requires
`asahi-uboot` and `grub-arm64-efi`).

If using audio, you must install `asahi-audio`, ensure the speakersafetyd
service is [enabled](/config/services/index.md#enabling-services), and set up
[pipewire and wireplumber](/config/media/pipewire.md).

Firmware can be updated with `asahi-fwupdate` from `asahi-scripts`. It is

> recommended to do so whenever the asahi-firmware package is updated.

Reviewer 2: This is not true.

`asahi-firmware` does not contain any firmware, all the firmware is extracted from MacOS. These scripts are used either in the initramfs, or for extracting the firmware from MacOS by the installer script

Author: Once again: the initramfs copies the files, but asahi-firmware does essentially what the install script does. It was recommended to me on irc than when asahi-fwupdate package is updated, this could signal that they are bringing in new firmware, hence the suggestion to run the command.

Reviewer 2: I don't think that it's possible to bring in new firmware, but if this is really the case, shouldn't it be run as a post install hook instead of making the user do it manually?

Author: Maybe my wording in unclear. I just meant that whenever that package is updated, the user may want to run that script. Running automatically by trigger may be also an option.

Author: See e.g. https://github.com/AsahiLinux/docs/wiki/Open-OS-Ecosystem-on-Apple-Silicon-Macs#packaged-blobs (as well as some irc conversation I cannot find now): there is definitely a possibility that new blobs are going to be used, and when this is the case, it will be implemented through an update of asahi-installer/asahi-firmware pkg. So maybe not all updates will require running fwupdate, but it is still advisable to do so.

Author: This is the ref to irc: https://oftc.irclog.whitequark.org/asahi/2023-04-11#32057960;

> the used OS would be responsible for providing asahi-fwextract and running it on upgrades 

I am happy to either recommend it (as is now) or use a trigger.

Reviewer: pretty sure jannau meant *macOS* upgrades, not upgrades of a package on the linux side, if the script extracts fw from macOS

Author: If you read the lines a bit above and below the linked one, it is already explained that they do _not_ mean upgrades of macos. The rationale is that they change/upgrade `asahi-installer` (which is where we get `asahi-firmware`) whenever they discover they need to provide additional firmware in `/boot/vendorfw/firmware.{cpio,tar.gz}`.
